### PR TITLE
CLI interface and Read depth option

### PR DESCRIPTION
**Pull request by madbonez**
I made cli interface for my own purposes I hope it can help for someone. Please read my code and get it in main branch 🙂  I can change something if you want. It'll be great if you adopt my changes because i don't want to see multiple npm packages that do almost same things 😭  
you can run it now by this command in your command line. directly from my repo
`npx madbonez/node-directory-tree#read-depth`

all my change is totally optional with backward compatibility. all tests green

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -186,3 +188,41 @@ Make sure you have the dev dependencies installed (e.g. `npm install .`)
...
+
+### Available options
+-p, --path string      🗂 The input folder to process. Required.                                     
+-e, --exclude string   🐒 Exclude some folders from processing by regexp string. Ex -e "test_data/some_dir$|js|.DS_Store"                                            
+-o, --output string    📝 Put result into file provided by this options. Overwrites if exists.       
+-d, --depth number     ☞ Reads dirs in deep as specified. Usage of size attribute with depth option is prohibited.                                                                
+--attributes string    ℹ️ Grab file attributes. Example: --attributes size,type,extension. Usage of size attribute with depth option is prohibited                                
+--pretty               💎 Json pretty print               
+
+## CLI usage
```
Comment (mihneadb) [inline README.md:211]: This ended up written twice.

File: bin/index.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+}
+
+if (Object.keys(options).length === 0 || options.help || !options.path) {
+    console.log(usage)
+    return;
+}
+
+if (!fs.existsSync(options.path)) {
+    console.log('-----------------------------------------------------------------------------------------------------')
+    console.log(`ERROR: Folder \"${options.path}\" doesn\'t exist please check your args`);
```
Comment (mihneadb) [inline bin/index.js:79]: `doesn't exist. Please check your args` or `doesn't exist; please check your args` pls

**Review by mihneadb — CHANGES_REQUESTED** [review]
Hey, thanks so much for putting this together! There are just two minor changes I mentioned in the review, please update them and then I'll merge & release this.

**Comment by spacedawwwg** [comment]
+1 for this. Great work.

File: README.md (hunk 1 above)
Comment (madbonez) [reply README.md]: thx! fixed

**Review by mihneadb — APPROVED** [review]
🙏

**Comment by mihneadb** [comment]
@madbonez looks good! One last q (and sorry for not figuring it out in the beginning). Why not have `path` as an argument? In the spirit of all UNIX CLIs I'd say one expects to run `directory-tree /path/foo` and not `directory-tree --path /path/foo`. If you agree, let's pls add that change and then I'll merge it. 🙏  thanks!

File: bin/index.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+const commandLineUsage = require('command-line-usage')
+const commandLineArgs = require('command-line-args')
+const directoryTree = require('../lib/directory-tree');
+
+const optionList = [
+    {
+        name: 'path',
+        alias: 'p',
+        required: true,
+        defaultOption: true,
```
Comment (madbonez) [inline bin/index.js:13]: @mihneadb i totally agree with you about default argument! Fortunately, it easy  🙂

**Review by mihneadb — APPROVED** [review]

File: bin/index.js (hunk 3 above)
Comment (mihneadb) [reply bin/index.js:13]: Thanks!

**Comment by mihneadb** [comment]
@madbonez any chance you can take a look on https://github.com/mihneadb/node-directory-tree/issues/105 ? Thanks!

**Pull request merged**